.github/workflows/check-for-workflow-run.js

```diff
@@ -16,13 +16,18 @@ module.exports = async ({github, context, core, workflow_id, sha, ...config}) =>
   const runFilterDesc = sha ? `${workflow_id} on ${sha}` : workflow_id
   for (const run of result.data.workflow_runs) {
     if ((!sha || run.head_sha === sha)) {
-      if (run.conclusion === "success") {
-        // The SHA is fully tested, exit with success
-        console.log(`Last run of ${runFilterDesc} succeeded: ${run.html_url}`)
-        return
-      } else if (run.status === "failure" || run.status === "timed_out") {
-        core.setFailed(`Last run of ${runFilterDesc} did not succeed: ${run.html_url}`)
-        return
+      // The status property can be one of: “queued”, “in_progress”, or “completed”.
+      if (run.status === "completed") {
+        // The conclusion property can be one of:
+        // “success”, “failure”, “neutral”, “cancelled”, “skipped”, “timed_out”, or “action_required”.
+        if (run.conclusion === "success") {
+          // The SHA is fully tested, exit with success
+          console.log(`Last run of ${runFilterDesc} succeeded: ${run.html_url}`)
+          return
+        } else if (run.conclusion === "failure" || run.conclusion === "timed_out") {
+          core.setFailed(`Last run of ${runFilterDesc} did not succeed: ${run.html_url}`)
+          return
+        }
       }
```

Comment:
```suggestion
      if (run.status === "completed") {
        // The conclusion property can be one of:
        // “success”, “failure”, “neutral”, “cancelled”, “skipped”, “timed_out”, or “action_required”.
        if (run.conclusion === "success") {
          // The SHA is fully tested, exit with success
          console.log(`Last run of ${runFilterDesc} succeeded: ${run.html_url}`)
        } else if (run.conclusion === "failure" || run.conclusion === "timed_out") {
          core.setFailed(`Last run of ${runFilterDesc} did not succeed: ${run.html_url}`)
        } else {
          break
        }
        return
      }
```
More or less what I had in mind.

Reply:
Is having a single return statement something you desire in code? I'm not so familiar with that style, but I see it makes the code more like that written in languages that don't have statements.

I'll merge this as is because I feel the status quo of our codebase is to also use a multiple return statement coding style.